From 3d52bc1e758c37f3d815c6c598fe556ef4b7aef3 Mon Sep 17 00:00:00 2001
From: Andrea C G Mennucci <andrea.mennucci@gmail.com>
Date: Tue, 4 Feb 2020 19:22:19 +0100
Subject: [PATCH 2/9] add \ifdefined and \ifcsname

---
 plasTeX/Base/TeX/Primitives.py | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/plasTeX/Base/TeX/Primitives.py b/plasTeX/Base/TeX/Primitives.py
index bcd81d9..cc4e754 100644
--- a/plasTeX/Base/TeX/Primitives.py
+++ b/plasTeX/Base/TeX/Primitives.py
@@ -343,6 +343,26 @@ class ifcase(IfCommand):
         tex.processIfContent(tex.readNumber())
         return []
 
+class ifdefined(IfCommand):
+    args = 'name:Tok'
+    def invoke(self, tex):
+        a = self.parse(tex)
+        n = str(a['name'].macroName)
+        b = n in self.ownerDocument.context
+        tex.processIfContent(b)
+        return []
+
+class ifcsname(IfCommand):
+    def invoke(self, tex):
+        name = []
+        for t in tex:
+            if t.nodeType == Command.ELEMENT_NODE and t.nodeName == 'endcsname':
+                break
+            name.append(t)
+        n = ''.join(name)
+        b = n in self.ownerDocument.context
+        tex.processIfContent(b)
+        return []
 
 class let(Command):
     """ \\let """
-- 
2.17.1

