From ba8c8e61520c6c0718ed434bb0bdcae6aeefe265 Mon Sep 17 00:00:00 2001
From: Andrea C G Mennucci <andrea.mennucci@gmail.com>
Date: Sat, 1 Feb 2020 15:02:35 +0100
Subject: [PATCH 5/9] correctly parse \if clauses containing \newif

---
 plasTeX/TeX.py | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index d3434a2..a90c033 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -547,10 +547,15 @@ class TeX(object):
         cases = [[]]
         nesting = 0
         correctly_terminated = False
+        iterator = self.itertokens()
 
-        for t in self.itertokens():
+        for t in iterator:
             name = getattr(t, 'macroName', '') or ''
-            if name.startswith('if'):
+            if name == 'newif':
+                cases[-1].append(t)
+                cases[-1].append(next(iterator))
+                continue
+            elif name.startswith('if'):
                 cases[-1].append(t)
                 nesting += 1
             elif name == 'fi':
-- 
2.17.1

