From 50c2ed58d1ce0dd3a6391f0761868531d0746309 Mon Sep 17 00:00:00 2001
From: Andrea C G Mennucci <andrea.mennucci@gmail.com>
Date: Sat, 30 Nov 2019 18:41:45 +0100
Subject: [PATCH 8/9] correctly parse [{$]$}]

---
 plasTeX/TeX.py | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index 8794ebe..dd4f30e 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -873,13 +873,21 @@ class TeX(object):
             if t.catcode != Token.CC_ESCAPE and \
                (t == begin or str(t) == str(begin)):
                 level = 1
+                group = 0
                 for t in tokens:
                     source.append(t)
-                    if t.catcode != Token.CC_ESCAPE and \
+                    if t.catcode == Token.CC_BGROUP:
+                        group += 1
+                    elif t.catcode == Token.CC_EGROUP:
+                        group -= 1
+                        if group < 0:
+                            log.warning('optional argument contains too many end groups %r' % source)
+                            group = 0
+                    elif t.catcode != Token.CC_ESCAPE and group == 0 and \
                        (t == begin or str(t) == str(begin)):
                         toks.append(t)
                         level += 1
-                    elif t.catcode != Token.CC_ESCAPE and \
+                    elif t.catcode != Token.CC_ESCAPE and group == 0 and \
                          (t == end or str(t) == str(end)):
                         level -= 1
                         if level == 0:
-- 
2.17.1

