From f1794c77698e7e5fdb9a9a0c559dd944bd083f03 Mon Sep 17 00:00:00 2001
From: Andrea C G Mennucci <andrea.mennucci@gmail.com>
Date: Wed, 13 Nov 2019 10:50:42 +0100
Subject: [PATCH 03/10] add Tokenizer to input()

---
 plasTeX/TeX.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index df2de62..e612081 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -139,7 +139,7 @@ class TeX(object):
                 self.input(file)
                 self.jobname = os.path.basename(os.path.splitext(file.name)[0])
 
-    def input(self, source):
+    def input(self, source, Tokenizer=Tokenizer):
         """
         Add a new input source to the stack
 
-- 
2.32.0

