From 0c57a8c43f55d027bb84c7be89c9e87939cb814b Mon Sep 17 00:00:00 2001
From: "Andrea C. G. Mennucci" <andrea.mennucci@gmail.com>
Date: Wed, 2 Feb 2022 18:52:22 +0100
Subject: [PATCH 09/10] make sure that there are spaces in front of arguments
 without {

---
 plasTeX/TeX.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index 7efd758..9de6587 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -734,6 +734,8 @@ class TeX(object):
             # Get a TeX token (i.e. {...})
             if spec is None:
                 toks, source = self.readToken(expanded, parentNode=parentNode)
+                if source and source[0] in (string.ascii_letters+string.digits):
+                    source = ' ' + source
 
             # Get a single character argument
             elif len(spec) == 1:
-- 
2.32.0

