From 8a1f72f08b23c7af46ab8113fea867554f5d432f Mon Sep 17 00:00:00 2001
From: Andrea C. G. Mennucci <andrea.mennucci@gmail.com>
Date: Sat, 19 Feb 2022 22:55:34 +0100
Subject: [PATCH] do not reinsert paths in sys.path when they are already there

---
 plasTeX/Context.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/plasTeX/Context.py b/plasTeX/Context.py
index b92df0e..5227631 100644
--- a/plasTeX/Context.py
+++ b/plasTeX/Context.py
@@ -429,7 +429,8 @@ class Context(object):
             pypath = (path/module).with_suffix('.py')
             if not pypath.exists():
                 continue
-            sys.path.insert(0, str(path))
+            if str(path) not in sys.path:
+                sys.path.insert(0, str(path))
             spec = find_spec(module)
             if spec is None:
                 sys.path = orig_sys_path
@@ -454,7 +455,9 @@ class Context(object):
                 sys.path = orig_sys_path
                 plugin_module = import_module(plugin)
                 assert plugin_module.__file__
-                sys.path.insert(0, str(Path(plugin_module.__file__).parent.parent))
+                p_ = str(Path(plugin_module.__file__).parent.parent)
+                if p_ not in sys.path:
+                    sys.path.insert(0, p_)
                 try:
                     imported = import_module(plugin + '.Packages.' + module)
                     break
@@ -470,7 +473,9 @@ class Context(object):
 
         if imported is None:
             # Now try builtin plasTeX packages
-            sys.path.insert(0, str(Path(__file__).parent.parent))
+            p_ = str(Path(__file__).parent.parent)
+            if p_ not in sys.path:
+                sys.path.insert(0, p_)
             try:
                 imported = import_module('plasTeX.Packages.' + module)
             except (ImportError, ModuleNotFoundError) as msg:
--
libgit2 1.1.0

