From bec05d8186be368cbe0597e5cfc2d8bd699b6788 Mon Sep 17 00:00:00 2001
From: "Andrea C. G. Mennucci" <andrea.mennucci@gmail.com>
Date: Sat, 12 Mar 2022 09:52:21 +0100
Subject: [PATCH 1/4] TeX record main input file

---
 plasTeX/TeX.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index 7959e07..51b67e4 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -122,6 +122,9 @@ class TeX(object):
                 or encoding is utf-8, make encoding utf_8_sig.
                 otherwise, use the specified encoding.
                 '''
+
+                self.main_input_file = file
+
                 try:
                     encoding = self.ownerDocument.config['files'].get('input-encoding', 'utf_8_sig')
                 except (KeyError, AttributeError):
@@ -137,7 +140,10 @@ class TeX(object):
             # File object
             else:
                 self.input(file)
+                self.main_input_file = file.name
                 self.jobname = os.path.basename(os.path.splitext(file.name)[0])
+        else:
+            self.main_input_file = None
 
     def input(self, source, Tokenizer=Tokenizer):
         """
@@ -156,6 +162,9 @@ class TeX(object):
             elif isinstance(source, IOBase) and hasattr(source,'name'):
                 self.jobname = os.path.basename(os.path.splitext(source.name)[0])
 
+        if self.main_input_file is None and isinstance(source, IOBase) and hasattr(source,'name'):
+            self.main_input_file = source.name
+
         t = Tokenizer(source, self.ownerDocument.context)
         self.inputs.append((t, iter(t)))
         self.currentInput = self.inputs[-1]
-- 
2.32.0

