From b8dae4f823e36b1c320b645cb14db293f2bce0fc Mon Sep 17 00:00:00 2001
From: "Andrea C. G. Mennucci" <andrea.mennucci@gmail.com>
Date: Sat, 12 Mar 2022 09:53:41 +0100
Subject: [PATCH 2/4] kspewhich: local searches are wrt the main input

---
 plasTeX/TeX.py | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index 51b67e4..1968080 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -1345,13 +1345,12 @@ class TeX(object):
         # When, for example, ``\Input{name}`` is encountered, we should look in
         # the directory containing the file being processed. So the following
         # code adds the directory to the start of $TEXINPUTS.
-        TEXINPUTS = None
-        try:
-            srcDir = os.path.dirname(self.filename)
-        except AttributeError:
-            # I think this happens only for the command line file.
-            pass
-        else:
+        #
+        TEXINPUTS = srcDir = None
+
+        srcDir = None if self.main_input_file is None else  os.path.dirname(self.main_input_file)
+
+        if srcDir is not None:
             TEXINPUTS = os.environ.get("TEXINPUTS",'')
             t = TEXINPUTS.split(os.path.pathsep)
             if srcDir not in t:
-- 
2.32.0

