From 13c4eb08374289e5c8d6c4a0b79339b15b697c54 Mon Sep 17 00:00:00 2001
From: "Andrea C. G. Mennucci" <andrea.mennucci@gmail.com>
Date: Mon, 21 Feb 2022 19:38:18 +0100
Subject: [PATCH 103/103] kpsewhich faster for local files

---
 plasTeX/TeX.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index 89e5b8f..530832d 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -1321,7 +1321,7 @@ class TeX(object):
 
         return dictarg
 
-    def kpsewhich(self, name):
+    def kpsewhich(self, name, shortcircuit=True):
         """
         Locate the given file using kpsewhich
 
@@ -1340,6 +1340,12 @@ class TeX(object):
         
         srcDir = None if self.main_input_file is None else  os.path.dirname(self.main_input_file)
 
+        if shortcircuit and srcDir is not None:
+            n = name if os.path.isabs(name) else os.path.join(srcDir,name)
+            for j in (n, n+'.tex', n+'.sty'):
+                if os.path.isfile(j):
+                    return j
+            
         if srcDir is not None:
             TEXINPUTS = os.environ.get("TEXINPUTS",'')
             os.environ["TEXINPUTS"] = "%s%s%s%s" % (srcDir, os.path.pathsep, TEXINPUTS, os.path.pathsep)
-- 
2.32.0

