From ce398942ec3893b6bc9b2679f1e49ce5b7b6a3fd Mon Sep 17 00:00:00 2001
From: "Andrea C. G. Mennucci" <andrea.mennucci@gmail.com>
Date: Sat, 12 Mar 2022 09:54:53 +0100
Subject: [PATCH 3/4] kpsewhich faster for local files

---
 plasTeX/TeX.py | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/plasTeX/TeX.py b/plasTeX/TeX.py
index 1968080..ce3cbb9 100644
--- a/plasTeX/TeX.py
+++ b/plasTeX/TeX.py
@@ -1331,7 +1331,7 @@ class TeX(object):
 
         return dictarg
 
-    def kpsewhich(self, name):
+    def kpsewhich(self, name, shortcircuit=True):
         """
         Locate the given file using kpsewhich
 
@@ -1350,6 +1350,12 @@ class TeX(object):
 
         srcDir = None if self.main_input_file is None else  os.path.dirname(self.main_input_file)
 
+        if shortcircuit and srcDir is not None:
+            n = name if os.path.isabs(name) else os.path.join(srcDir,name)
+            for j in (n, n+'.tex', n+'.sty'):
+                if os.path.isfile(j):
+                    return j
+
         if srcDir is not None:
             TEXINPUTS = os.environ.get("TEXINPUTS",'')
             t = TEXINPUTS.split(os.path.pathsep)
-- 
2.32.0

